spark/src/main/scala/io/delta/sharing/spark/DeltaSharingClient.scala

```diff
@@ -333,15 +348,31 @@ private[spark] class DeltaSharingRestClient(
     }
   }
 
+  private[sharing] def getHttpHeaders(profile: DeltaSharingProfile): Map[String, String] = {
+    // Load the custom http header provider and get custom headers if there are any.
+    val customHttpHeadersProviderClass =
+      SparkSession.active.sessionState.conf
+        .getConfString("spark.delta.sharing.customhttpheaders.provider.class",
+        "io.delta.sharing.spark.BaseCustomHttpHeadersProvider")
+    val customeHttpHeadersProvider =
+      Class.forName(customHttpHeadersProviderClass)
+        .getConstructor()
+        .newInstance()
+        .asInstanceOf[CustomHttpHeadersProvider]
+    Map(
+      HttpHeaders.AUTHORIZATION -> s"Bearer ${profile.bearerToken}",
+      HttpHeaders.USER_AGENT -> DeltaSharingRestClient.USER_AGENT
+    ) ++ customeHttpHeadersProvider.getHeaders
```

Comment:
Should we check that the custom headers do not conflict with some important headers?

Reply:
[HTTP RFC](https://stackoverflow.com/questions/4371328/are-duplicate-http-response-headers-acceptable) says multiple headers with the same key is allowed. Let's conform with the standard and allow the user to do whatever is legal.

Reply:
When you combine maps like that in Scala, the values get overridden by the custom headers.

How does the Delta Sharing server handle multiple AUTHORIZATION headers?

Reply:
Ok -- added check that authz and user-agent can not be overridden.